effects/sleep.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+import time
+
+import click
+
+from rpi_ws281x import PixelStrip, Color
+
+# LED strip default configuration:
+#LED_COUNT = 50       # Number of LED pixels.
+LED_PIN = 18          # GPIO pin connected to the pixels (18 uses PWM!).
+LED_FREQ_HZ = 800000  # LED signal frequency in hertz (usually 800khz)
+LED_DMA = 10          # DMA channel to use for generating signal (try 10)
+LED_BRIGHTNESS = 255  # Set to 0 for darkest and 255 for brightest
+LED_INVERT = False    # True to invert the signal (when using NPN transistor level shift)
+LED_CHANNEL = 0       # set to '1' for GPIOs 13, 19, 41, 45 or 53
+
+@click.command()
+@click.option('-l', '--leds', type=int, default=50, show_default=True, help='Amount of LEDs in the string')
+@click.option('-s', '--speed', type=float, default=1.0, show_default=True, help='Speed of the effect')
+@click.option('-c', '--color', type=str, default='FF0000', show_default=True, help='RGB Value for the effect. You need to provide this value as a hexadecimal.')
+def main(leds: int, speed: float, color: str):
+    '''
+    This is a sleep effect which provide a light for the night
+    '''
+    if not color.startswith('0x'):
+        color = '0x' + color
+        
+    print(f'Sleep effect started with following parameter: leds={leds}, speed={speed}, color={color}')
+    color_dict = parseColorString(color)
+    
+    strip = PixelStrip(leds, LED_PIN, LED_FREQ_HZ, LED_DMA, LED_INVERT, LED_BRIGHTNESS, LED_CHANNEL)
+    strip.begin() # Intialize the library (must be called once before other functions).
+    
+    runEffect(strip, speed, color_dict['red'], color_dict['green'], color_dict['blue']);
+    pass
+
+def parseColorString(color: str):
+    '''
+    Parse the color given as string to single RGB colors as int values and return the dictionary with
+    parsed values
+    '''
+    color_int_val = int(color, base=16)
+    red_mask = 0xFF0000
+    green_mask = 0x00FF00
+    blue_mask = 0x0000FF
+    
+    return {
```

Comment:
If we shift first, we need only a mask for the first 8 bit and not for every color

Reply:
`
eight_bit_mask = 0xFF

return {
    'red': (color_int_val >> 16) & eight_bit_mask ,
    'green': (color_int_val >> 8) & eight_bit_mask,
    'blue': color_int_val & eight_bit_mask
}
`